robots/cmd/perf-report-creator/main.go

```diff
@@ -0,0 +1,319 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"k8s.io/apimachinery/pkg/util/errors"
+	. "kubevirt.io/project-infra/robots/pkg/flakefinder"
+)
+
+type options struct {
+	since              time.Duration
+	performanceJobName string
+	vmMetricsList      string
+	vmiMetricsList     string
+	outputDir          string
+}
+
+func flagOptions() options {
+	o := options{}
+	flag.DurationVar(&o.since, "since", 24*7*time.Hour, "Filter the periodic job in the time window")
+	flag.StringVar(&o.performanceJobName, "performance-job-name", "periodic-kubevirt-e2e-k8s-1.25-sig-performance", "usuage, name of the performance job for which data is collected")
+	flag.StringVar(&o.vmMetricsList, "vm-metrics-list", string(ResultTypeVMICreationToRunningP95), "comma separated list of metrics to be extracted for vms")
+	flag.StringVar(&o.vmiMetricsList, "vmi-metrics-list", string(ResultTypeVMICreationToRunningP95), "comma separated list of metrics to be extracted for vmis")
+	flag.StringVar(&o.outputDir, "output-dir", "output", "the output directory were json data will be written")
+	flag.Parse()
+	return o
+}
+
+func main() {
+	opts := flagOptions()
+
+	ctx := context.Background()
+	storageClient, err := storage.NewClient(ctx)
+	if err != nil {
+		// return fmt.Errorf("Failed to create new storage client: %v.\n", err)
+	}
+
+	allJobs, err := listAllJobs(ctx, storageClient)
```

Comment:
And looking further into this, it seems like you assume that the content will be traversed recursively which I think is not the case. I think you need to descend into each directory that you want to traverse.

Or my gcs lib knowledge is not up to date, which may well be :smile:

Reply:
yes, I need to update the url to `gs://kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.25-sig-performance/` I'll push an update for that with a todo to improve the performance of the `since` part

Although I do make a subsequent call to download the build-log.txt at

```
gs://kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.25-sig-performance/<job-id>/build-log.txt
```

Does the above appropriately handle what is expected from the api?
> And looking further into this, it seems like you assume that the content will be traversed recursively which I think is not the case. I think you need to descend into each directory that you want to traverse.

Reply:
@dhiller pushed a fix as suggested with a todo for sorting enhancement. PTAL